bot/utils/transformData.ts

```diff
@@ -0,0 +1,96 @@
+import { poolpairs, stats } from '@defichain/whale-api-client';
+import { BigNumber } from 'bignumber.js';
+
+import { StateRelayer } from '../../generated';
+import { MasterNodeData, PairData, VaultData } from './types';
+
+
+const DECIMALS = 18;
+const DENOMINATION = 'USDT';
+
+type AsyncReturnType<T extends (...args: any) => any> = T extends (...args: any) => Promise<infer R> ? R : any;
+type StatsData = AsyncReturnType<typeof stats.Stats.prototype.get>;
+type PoolPairData = AsyncReturnType<typeof poolpairs.PoolPairs.prototype.list>;
+type DexPriceResult = AsyncReturnType<typeof poolpairs.PoolPairs.prototype.listDexPrices>;
```

Comment:
For these types, I believe you can get them directly from `dist` folder, can you check?
```
import { DexPricesResult, PoolPairData } from '@defichain/whale-api-client/dist/api/poolpairs';
import { StatsData } from '@defichain/whale-api-client/dist/api/stats';
```